components/ActivityForm/index.js

```diff
@@ -0,0 +1,155 @@
+import styled from "styled-components";
+import { StyledButton } from "@/components/StyledButton";
+import { useRouter } from "next/router";
+import { StyledBackLink } from "@/components/StyledLinks";
+import { Icon } from "@/components/Icons";
+
+export default function ActivityForm({ onAddActivity }) {
+  const router = useRouter();
+
+  function handleSubmit(event) {
+    event.preventDefault();
+    const formData = new FormData(event.target);
+    const newActivity = Object.fromEntries(formData);
+    newActivity.category = formData.getAll("category");
+    if (newActivity.category.length === 0) {
+      alert("Please select at least one category.");
+      return false;
+    }
+    onAddActivity(newActivity);
+    router.push("/");
+  }
+
+  return (
+    <StyledForm onSubmit={handleSubmit}>
+      <StyledBackLink href={"/"}>
+        <Icon name="chevronLeft" />
+        Back to all Activities
+      </StyledBackLink>{" "}
+      <StyledLabel htmlFor="title">Activity</StyledLabel>
+      <StyledInput
+        id="title"
+        name="title"
+        type="text"
+        maxLength="150"
+        placeholder="add activity-title"
+        required
+      />
+      <StyledLabel htmlFor="area">Area</StyledLabel>
+      <StyledInput
+        id="area"
+        name="area"
+        type="text"
+        placeholder="add area"
+        required
+      />
+      <StyledLabel htmlFor="country">Country</StyledLabel>
+      <StyledInput
+        id="country"
+        name="country"
+        type="text"
+        placeholder="add country"
+        required
+      />
+      <StyledLabel htmlFor="category">Category</StyledLabel>
+      <StyledCheckboxContainer>
+        <label htmlFor="outdoor">
```

Comment:
I think the 'htmlFor' isn't necessary as long as you wrap the label around the input. This structure makes clear what the label is for.